### PR TITLE
Feature: Added large tab size to react-tabs

### LINKED ISSUE #25551: Update TabList/Tab to support new large size

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Current Behavior

Tabs support small and medium sizes

## New Behavior

Tabs support small, medium, and large sizes

## Changes
- added large to the size property union
- Updated tab style hook to set large styles according to figma
- Added large size to storybook
- Updated tab tests to cover large size
- Updated vr-tests to include large size

## Related Issue(s)

Fixes #25551

## Screenshots

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/Avatar-mount_base.html">1318</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/Avatar-mount.html">1307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/Button-mount_base.html">957</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/Button-mount.html">941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProvider-mount_base.html">1579</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProvider-mount.html">1578</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-mount_base.html">645</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-mount.html">635</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">583</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">595</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">625</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/MakeStyles-mount_base.html">1904</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/MakeStyles-mount.html">1907</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/SpinButton-mount_base.html">2595</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25577/perf-test-react-components/SpinButton-mount.html">2585</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `79.284 kB`<br />`24.446 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `84.771 kB`<br />`21.265 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.956 kB`<br />`13.883 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `14.996 kB`<br />`6.013 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `63.534 kB`<br />`17.979 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `22.646 kB`<br />`7.228 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `23.536 kB`<br />`7.512 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |    `24.096 kB`<br />`7.09 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |    `36.85 kB`<br />`9.917 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `43.875 kB`<br />`11.134 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `39.544 kB`<br />`10.816 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |   `46.99 kB`<br />`12.201 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |    `52.535 kB`<br />`11.4 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `71.863 kB`<br />`20.939 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `67.641 kB`<br />`19.897 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `8.607 kB`<br />`3.622 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `9.782 kB`<br />`3.998 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |    `8.708 kB`<br />`3.678 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `76.153 kB`<br />`24.665 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `75.885 kB`<br />`24.635 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.941 kB`<br />`17.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.948 kB`<br />`52.993 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `83.147 kB`<br />`24.799 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `16.505 kB`<br />`5.925 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |   `10.826 kB`<br />`4.286 kB` |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `112.443 kB`<br />`34.387 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `24.428 kB`<br />`7.899 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |    `9.384 kB`<br />`3.883 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `11.852 kB`<br />`4.888 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          |   `117.131 kB`<br />`36.2 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             |   `120.2 kB`<br />`36.726 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `11.004 kB`<br />`4.188 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |  `54.074 kB`<br />`15.269 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `103.672 kB`<br />`31.871 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `10.495 kB`<br />`3.851 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |   `19.826 kB`<br />`7.417 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/Progress.fixture.js'>Progress</abbr>                                                                              |   `13.526 kB`<br />`5.113 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `15.807 kB`<br />`5.909 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `36.436 kB`<br />`12.126 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.294 kB`<br />`5.723 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `23.255 kB`<br />`7.954 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `32.108 kB`<br />`10.195 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `45.826 kB`<br />`12.858 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `20.003 kB`<br />`6.449 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |  `33.443 kB`<br />`10.585 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `11.828 kB`<br />`4.627 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `15.138 kB`<br />`5.064 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `25.687 kB`<br />`8.331 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/SingleThemeToken.fixture.js'>Single theme token import</abbr>                                                        |            `69 B`<br />`89 B` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>                                                                  |    `29.65 kB`<br />`6.444 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr>                                                                |   `17.486 kB`<br />`5.057 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `42.032 kB`<br />`14.739 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6ab9ad73f61226c779bd6bc297daa12b11791551'>6ab9ad73f61226c779bd6bc297daa12b11791551</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-theme/src/global/typographyStyles.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const typographyStyles: TypographyStyles = {
   body2: {
     fontFamily: tokens.fontFamilyBase,
     fontSize: tokens.fontSizeBase400,
-    fontWeight: tokens.fontWeightSemibold,
+    fontWeight: tokens.fontWeightRegular,
```
Comment (smhigley) [inline packages/react-components/react-theme/src/global/typographyStyles.ts:29]: This seems right to me and I'm guessing you already talked with design, I just wanted to double check that this was the desired weight for other body2 uses, e.g. Miro's PR for body2 (#24378) references adding a react-text Body2 component, though it looks like that hasn't happened yet.

**Review by smhigley — APPROVED** [review]

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts [hunk 2]
```diff
@@ -414,6 +471,7 @@ export const useTabStyles_unstable = (state: TabState): TabState => {
     state.contentReservedSpaceClassName = mergeClasses(
       reservedSpaceClassNames.content,
       contentStyles.base,
+      size === 'large' && contentStyles.large,
```
Comment (smhigley) [inline packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts:474]: Sorry, one change I just caught -- this looks like it should actually be `contentStyles.largeSelected`, and needs to come after the `contentStyles.selected` classname below, so the large font-size overrides the regular selected font-size

**Review by smhigley — CHANGES_REQUESTED** [review]

File: packages/react-components/react-theme/src/global/typographyStyles.ts (hunk 1 above)
Comment (GeoffCoxMSFT) [reply packages/react-components/react-theme/src/global/typographyStyles.ts:29]: (as mentioned in chat) Yes.  This is a code bug.  Probably a cut-and-paste bug as the figma always said regular font weight.  Yvonne and Daisy agree.

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts (hunk 2 above)
Comment (GeoffCoxMSFT) [reply packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts]: Great catch!

**Comment by GeoffCoxMSFT** [comment]
Force merging as this is react-tabs feature and coastal approved.

**Pull request merged**